github_cog.py

```diff
@@ -0,0 +1,111 @@
+import discord
+import github.GithubException
+from discord.ext import commands
+from github import Github
+from contextlib import closing
+import studybot
+
+
+# noinspection PyBroadException
+class GitHub_Integration(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name='github-join', help=studybot.help_dict.get('github-join'))
+    async def join(self, ctx, github_name='null'):
+        # Make sure they included the github_name arg, and send a response.
+        if github_name == 'null':
+            msg = 'You need to include your GitHub Username like this: '
+            msg += '`!github-join StudyBot`'
+            await ctx.message.reply(msg)
+            return
+        else:
+            g = Github(studybot.TOKENS.get('GITHUB_API_KEY'))
```

Comment:
Good point on 1, although I'm pretty sure it would break when it tries to add it to the TOKENS dictionary in the first place. 

I'm following the recommendations of the pygithub docs for the var name.